### PR TITLE
Prevent spamming log output with RX statistics by default

### PR DESCRIPTION
**Pull request by niklaut**
This makes the reception loop reporting less verbose.

In my usage, I was getting spammed with `read failed: Resource temporarily unavailable` messages, however, `read()` returning EAGAIN is not an error when using the non-blocking API, thus it's now filtered away.
I also put the spammy `actual_read_count` reporting behind the CL option `--detailed_rx` analog to `--detailed_tx`. Sadly the short option `-R` was already taken, so I chose `-U`.

### CODE REVIEW

File: linux-serial-test.c [hunk 1]
```diff
@@ -291,6 +292,7 @@ static void display_help(void)
 			"  -d, --divisor            UART Baud rate divisor (can be used to set custom baud rates)\n"
 			"  -R, --rx_dump            Dump Rx data (ascii, raw)\n"
 			"  -T, --detailed_tx        Detailed Tx data\n"
+			"  -U, --detailed_rx        Detailed Rx data\n"
```
Comment (andy-shev) [inline linux-serial-test.c:295]: Yeah, logically I would swap U and R.

Comment (niklaut) [reply linux-serial-test.c:295]: I would prefer it too, but it breaks backward compatibility of the CLI? Is that ok?

Comment (niklaut) [reply linux-serial-test.c]: Ok, I changed the `--rx_dump` short flag to `-D`, which is at least a little more logical.

**Comment by cbrake** [comment]
@niklaut thanks for the improvement!

**Pull request merged**